docs/source/l1_to_l2b_e2e.rst

```diff
@@ -0,0 +1,187 @@
+        dark_path = os.path.join(processed_cal_path, "dark_current_20240322.fits")
+        flat_path = os.path.join(processed_cal_path, "flat.fits")
+        fpn_path = os.path.join(processed_cal_path, "fpn_20240322.fits")
+        cic_path = os.path.join(processed_cal_path, "cic_20240322.fits")
+        bp_path = os.path.join(processed_cal_path, "bad_pix.fits")
+
+The raw science data files and mock calibration files are defined.
+
+.. code-block:: python
+
+        l1_data_filelist = [os.path.join(l1_datadir, "{0}.fits".format(i)) for i in [90499, 90500]]
+        mock_cal_filelist = [os.path.join(l1_datadir, "{0}.fits".format(i)) for i in [90526, 90527]]
+        tvac_l2b_filelist = [os.path.join(l2b_datadir, "{0}.fits".format(i)) for i in [90529, 90531]]
+
+The calibration database (`caldb`) is initialized, and calibration entries are created. The following is mocking code for squeezing the test data in the official format. 
+
+.. code-block:: python
+
+        pri_hdr, ext_hdr = mocks.create_default_headers()
+        ext_hdr["DRPCTIME"] = time.Time.now().isot
+        ext_hdr['DRPVERSN'] = corgidrp.__version__
+        mock_input_dataset = data.Dataset(mock_cal_filelist)
+
+        this_caldb = caldb.CalDB()  # Connection to the calibration database
+
+Non-linearity calibration, KGain, noise maps, flat field, and bad pixel maps are generated and stored in the calibration database.
+
+.. code-block:: python
+
+        # Nonlinearity calibration
+        nonlin_dat = np.genfromtxt(nonlin_path, delimiter=",")
+        nonlinear_cal = data.NonLinearityCalibration(nonlin_dat, pri_hdr=pri_hdr, ext_hdr=ext_hdr,
+                                                     input_dataset=mock_input_dataset)
+        nonlinear_cal.save(filedir=l2b_outputdir, filename="mock_nonlinearcal.fits")
+        this_caldb.create_entry(nonlinear_cal)
+
+        # KGain
+        kgain_val = 8.7
+        kgain = data.KGain(np.array([[kgain_val]]), pri_hdr=pri_hdr, ext_hdr=ext_hdr,
+                           input_dataset=mock_input_dataset)
+        kgain.save(filedir=l2b_outputdir, filename="mock_kgain.fits")
+        this_caldb.create_entry(kgain)
+
+The ``walker.walk_corgidrp`` function is the main part of the pipeline responsible for transforming the raw L1 data into L2b data. This function applies all necessary calibration steps and generates the output files.
+
+The processed L2b data is compared against TVAC data to verify correctness.
+
+.. code-block:: python
+
+        new_l2b_filenames = [os.path.join(l2b_outputdir, "{0}.fits".format(i)) for i in [90499, 90500]]
+
+        for new_filename, tvac_filename in zip(new_l2b_filenames, tvac_l2b_filelist):
+            img = data.Image(new_filename)
+
+            with fits.open(tvac_filename) as hdulist:
+                tvac_dat = hdulist[1].data
+
+            diff = img.data - tvac_dat
+
+            assert np.all(np.abs(diff) < 1e-5)
+
+    if __name__ == "__main__":
```

Comment:
I don't think we need this. Think of it as something is copying and pasting each code snippet you provide and running it